xarray/tests/test_combine.py

```diff
@@ -742,10 +742,16 @@ def test_auto_combine(self):
             Dataset({"x": ("a", [0]), "y": ("a", [0])}),
             Dataset({"y": ("a", [1]), "x": ("a", [1])}),
         ]
+
         actual = auto_combine(objs)
         expected = Dataset({"x": ("a", [0, 1]), "y": ("a", [0, 1])})
         assert_identical(expected, actual)
 
+        objs = [Dataset({"x": [0], "y": [0]}), Dataset({"x": [0]})]
+        actual = auto_combine(objs)
+        expected = Dataset({"x": [0], "y": [0, np.nan]})
```

Comment:
This is interesting. I guess `y` is in `concat_over` so it gets filled. I wonder if filling in coordinate variables with NaNs is a good idea.

Reply:
This doesn't look right to me. I think coordinates/indexes should be excluded from auto-filling with NaN.

Reply:
Ok, I'll revert the change to the test and update the logic.